store/generated_key/memstore/memstore.go

```diff
@@ -57,110 +53,44 @@ var _ common.GeneratedKeyStore = (*MemStore)(nil)
 func New(
 	ctx context.Context, verifier *verify.Verifier, log logging.Logger, config *memconfig.SafeConfig,
 ) (*MemStore, error) {
-	store := &MemStore{
-		log:       log,
-		config:    config,
-		keyStarts: make(map[string]time.Time),
-		store:     make(map[string][]byte),
-		verifier:  verifier,
-		codec:     codecs.NewIFFTCodec(codecs.NewDefaultBlobCodec()),
-	}
-
-	if store.config.BlobExpiration() != 0 {
-		log.Info("memstore expiration enabled", "time", store.config.BlobExpiration)
-		go store.pruningLoop(ctx)
-	}
-
-	return store, nil
-}
-
-// pruningLoop ... runs a background goroutine to prune expired blobs from the store on a regular interval.
-func (e *MemStore) pruningLoop(ctx context.Context) {
-	timer := time.NewTicker(DefaultPruneInterval)
-
-	for {
-		select {
-		case <-ctx.Done():
-			return
-
-		case <-timer.C:
-			e.pruneExpired()
-		}
-	}
-}
-
-// pruneExpired ... removes expired blobs from the store based on the expiration time.
-func (e *MemStore) pruneExpired() {
-	e.mu.Lock()
-	defer e.mu.Unlock()
-
-	for commit, dur := range e.keyStarts {
-		if time.Since(dur) >= e.config.BlobExpiration() {
-			delete(e.keyStarts, commit)
-			delete(e.store, commit)
-
-			e.log.Debug("blob pruned", "commit", commit)
-		}
-	}
+	return &MemStore{
+		ephemeral_db.New(ctx, config, log),
+		log,
+		verifier,
+		codecs.NewIFFTCodec(codecs.NewDefaultBlobCodec()),
+	}, nil
 }
 
 // Get fetches a value from the store.
 func (e *MemStore) Get(_ context.Context, commit []byte) ([]byte, error) {
-	time.Sleep(e.config.LatencyGETRoute())
-	e.reads++
-	e.mu.RLock()
-	defer e.mu.RUnlock()
-
 	var cert verify.Certificate
 	err := rlp.DecodeBytes(commit, &cert)
 	if err != nil {
 		return nil, fmt.Errorf("failed to decode DA cert to RLP format: %w", err)
 	}
 
-	var encodedBlob []byte
-	var exists bool
-	if encodedBlob, exists = e.store[string(cert.BlobVerificationProof.InclusionProof)]; !exists {
-		return nil, fmt.Errorf("commitment key not found")
-	}
-
-	// Don't need to do this really since it's a mock store
-	err = e.verifier.VerifyCommitment(cert.BlobHeader.Commitment, encodedBlob)
+	// construct lookup key identical to EigenDA V1
+	encodedBlob, err := e.FetchEphemeralEntry(cert.BlobVerificationProof.InclusionProof)
```

Comment:
memntioned elsewhere but really think we should rename to just FetchEntry, again because ephemerality is a dynamic property (depends whether memstore was started with BlobDuration > 0 or not).

Reply:
[27bc9f6](https://github.com/Layr-Labs/eigenda-proxy/pull/305/commits/27bc9f6174c1498c61cc79b963ec52cb85b5ee88)